web-components/src/localization/index.ts

```diff
@@ -15,12 +16,27 @@ export const { getLocale, setLocale } = configureLocalization({
   targetLocales,
   loadLocale: (locale: string) => import(`./localization/locales/${locale}.js`),
 })
+
+// Wait for the locale to be set
+export let isLocaleSet = false
+
+/**
+ * Get the locale if it's set or delay 100ms to get it
+ */
+export const getLocaleAfterInit = async (): Promise<string> => {
+  // Delay to wait for the locale to be set
+  if (!isLocaleSet) {
+    await delay(100)
+  }
+  return getLocale()
```

Comment:
Ain't it better to be safe:

```suggestion
  while (!isLocaleSet) {
    await delay(100)
  }
  return getLocale()
```